server/cache.ts

```diff
@@ -0,0 +1,41 @@
+import mongoose from 'mongoose';
+
+// TODO: add typing
+
+class Cache {
+    name: string;
+    model: any;
+    query: any;
+    data: any;
+
+    constructor(name, model, query) {
+        this.name = name;
+        this.model = model;
+        this.query = query;
+    }
+
+    exec() {
+        this.query.exec()
+        .then((data) => {
+            this.data = data;
+            console.log("Finished caching " + this.name + " cache");
+        })
+        .catch((error) => {
+            console.error("ERROR: Failed to fetch Requests from MongoDB");
+        });
+    }
+
+    init() {
+        this.exec();
+        this.model.watch().on('change', this.exec);
+    }
+
+    getData() {
+        return this.data;
+    }
+}
+
+import { Request } from "./models/requestModel";
```

Comment:
one more nitpick: can we import at the top?

Reply:
done!